src/gc/env/gcenv.ee.h

```diff
@@ -78,7 +80,7 @@ class GCToEEInterface
 
     static void GcEnumAllocContexts(enum_alloc_context_func* fn, void* param);
 
-    static void AttachCurrentThread(); // does not acquire thread store lock
+    static bool CreateBackgroundThread(Thread** thread, GCBackgroundThreadFunction threadStart, void* arg);
```

Comment:
> The value is written there by the thread itself once it starts running

This is pretty odd contract. I think it would be better to move the waiting for the thread to be created into CreateBackgroundThread, so that this can just return `Thread *`.